### PR TITLE
Update professional plan with starter naming

### PR DESCRIPTION
**Pull request by jprcoliveira**
## Description

Closes https://github.com/camunda/developer-experience/issues/115.

## When should this change go live?

- [x] This change is not yet live and should not be merged until Sept 25, 2023
- [ ] There is no urgency with this change.
- [x] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by jprcoliveira** [comment]
@christinaausley we need to create some redirects for the URLs below (this will be for next and 8.2. only). Could you help me with those? If you have some how-to guides I can give it a try myself :) 

- components/console/manage-plan/upgrade-to-professional-plan/ —> components/console/manage-plan/upgrade-to-starter-plan/
 
- components/console/manage-plan/cancel-professional-subscription —> components/console/manage-plan/cancel-starter-subscription

**Comment by akeller** [comment]
@jprcoliveira can you put the hold label back so we don't accidentally merge and release this?

**Comment by jprcoliveira** [comment]
> @jprcoliveira can you put the hold label back so we don't accidentally merge and release this?

Done and apologies :)

**Comment by christinaausley** [comment]
Sure thing @jprcoliveira 👍 Planning to have a look at this tomorrow.

File: docs/components/concepts/clusters.md [hunk 1]
```diff
@@ -6,20 +6,20 @@ description: "Learn more about the clusters available in your Camunda Platform 8
 
 A cluster is a provided group of nodes that run Camunda Platform 8. By default, Camunda Platform 8 clusters are production-ready.
 
-Professional and Enterprise plan customers can create as many production or development clusters as they want.
+Starter and Enterprise plan customers can create production or development within the limits of their subscription.
```
Comment (akeller) [inline docs/components/concepts/clusters.md:9]: Is there a link to the marketing website for more subscription info?

**Comment by akeller** [comment]
This is looking good so far 💪 

@christinaausley can you have a look at the merge conflicts here to keep @jprcoliveira unblocked?

**Comment by jprcoliveira** [comment]
> Nice work @jprcoliveira -- I've pushed in some redirects and will make sure those are working today.
> 
> One question -- in what cases do we include `(formerly Professional)` and for how long would we like this to remain in the docs?

@christinaausley I kept it only once in one of the docs. Do you have any recommendations? Shall we do it across the board? I would keep it for 6 months or so until we potentially migrate everyone over.

File: docs/components/concepts/clusters.md (hunk 1 above)
Comment (jprcoliveira) [reply docs/components/concepts/clusters.md:9]: @akeller we could use the link to the limits that will be something like https://camunda.com/legal/fair-usage-limits-starter-plan. WDYT?

**Comment by christinaausley** [comment]
> @christinaausley I kept it only once in one of the docs. Do you have any recommendations? Shall we do it across the board? I would keep it for 6 months or so until we potentially migrate everyone over.

I'm okay with keeping this in the frequented "plan" docs you've added this to. Can revisit to remove 👍

**Comment by jprcoliveira** [comment]
@christinaausley and @akeller we will go live with these changes around 12pm (Berlin time) on Monday (Sept 25th, 2023) so we can go live with these changes any time after that. Let me know if you need anything else from my side.

File: docs/components/concepts/clusters.md (hunk 1 above)
Comment (akeller) [reply docs/components/concepts/clusters.md:9]: Whoops, I missed this. I'll add this link. It goes live on Sept 25, right?

Comment (akeller) [reply docs/components/concepts/clusters.md:9]:
```suggestion
Starter and Enterprise plan customers can create production or development [within the limits of their subscription](https://camunda.com/legal/fair-usage-limits-starter-plan).
```

**Comment by akeller** [comment]
> @christinaausley and @akeller we will go live with these changes around 12pm (Berlin time) on Monday (Sept 25th, 2023) so we can push these changes live any time after that. Let me know if you need anything else from my side.

I will resolve any merge conflicts and merge this today at my EOD so it's in [stage.docs.camunda.io ](https://stage.docs.camunda.io/). Have a look at that Monday morning to make sure there are no additional changes you want to make immediately. If you have them, PR them, and we'll merge them ASAP when we sign on (around 8-9AM EST, 2-3PM Berlin time). 

If you want the changes in the docs earlier than that, please reach out ASAP (DM via Slack preferred) so I know to log on earlier on Monday.

**Review by akeller — APPROVED** [review]

**Pull request merged**